### DISCUSSION TITLE
Peculiar error, literally

### DISCUSSION THREAD

**Discussion by magthe** [emacs-openai/openai#29] [Q&A] [answered]
I've seen this a few times now. I have a fairly long-lived session open (a few hours at least, I'm not sure if I can find the exact time somehow), and then, all of a sudden I get this message about a "peculiar error" and the session is unusable.

```
[error] request--callback: peculiar error: 400
400 - Bad request.  Please check error message and your parameters
```

Anyway, is this something you recognise?

The above is everything that's put into `*Messages*` and I'm not sure if there's some way I can get more information somewhere. A debug mode for `openai` or `chatgpt` maybe?

**Comment by magthe** [comment]
I'm not sure if it's significant, but this is the information printed before the prompt that caused the "peculiar error: 400".

![screenshot_2023-09-19_16-11-06_221680268](https://github.com/emacs-openai/openai/assets/327270/8c4fb19b-630b-4fbf-973f-8db38cebdbf4)

**Comment by jcs090218** [comment]
Can you try our debugging section https://github.com/emacs-openai/openai#-debugging?

It should print out the reason why it didn't work!

**Comment by magthe** [comment]
OK, it's turned on. I'll post more information when I've gotten the error again.

**Comment by magthe** [comment]
This is the beginning of the error message:

```
[ERROR]: #s(request-response 400 nil ((error (message . This model's maximum context length is 4097 tokens. However, you requested 4114 tokens (2114 in the messages, 2000 in the completion). Please reduce the length of the messages or completion.) (type . invalid_request_error) (param . messages) (code . context_length_exceeded))) (error http 400) error https://api.openai.com/v1/chat/completions nil...
```

**Comment by jcs090218** [comment, marked as answer]
Duplicate of https://github.com/emacs-openai/codegpt/issues/9.

Try lower `chatgpt-max-tokens` for [chatgpt](https://github.com/emacs-openai/chatgpt). `codegpt-max-tokens` for [codegpt](https://github.com/emacs-openai/codegpt).
